storage/watchable_store.go

```diff
@@ -272,7 +277,13 @@ func (s *watchableStore) syncWatchings() {
 		}
 		// switch to tracking future events if needed
 		if nextRev > curRev {
-			s.synced[string(w.key)] = append(s.synced[string(w.key)], w)
+			k := string(w.key)
```

Comment:
the same code logic repeats twice. Can we make this a nice helper function instead?